src/ini.cr

```diff
@@ -71,6 +71,12 @@ class INI
 
   # Appends INI data to the given IO.
   def self.build(io : IO, ini, space : Bool = false) : Nil
+    # An empty section have to be at first, to prevent being included in another one.
+    if ini.is_a? Hash
+      ini.delete("").try &.each do |key, value|
```

Comment:
This is wrong! You are modifying data the user is passing.

What was wrong with the previous approach?

Reply:
I think the safest way to do this is to iterate `ini` twice, once looking for an empty section, once looking for non-empty sections.

Reply:
Right, yes